public-ui/src/app/data-browser/components/survey-answer/survey-answer-react.component.tsx

```diff
@@ -0,0 +1,361 @@
+    '#EE6857', '#5E869E', '#93003A', '#93003A', '#DF4A53'
+];
+
+const eightteenColors = [
+    '#2F4B7C', '#FA9659', '#BFE1C6', '#D2364F', '#AB0A42', '#6F98A0', '#3A5A86', '#93B8AC', '#FBAF57',
+    '#527997', '#F57D58', '#46698F', '#EC6556', '#C02049', '#60889F', '#80A8AA', '#E14D53', '#93003A',
+];
+
+const twentyFiveColors = [
+    '#00429D', '#93C4D2', '#6492C0', '#B61A49', '#E37B7E', '#FBAF57', '#73A2C6', '#FA9659', '#4771B2',
+    '#DF6772', '#A5D5D8', '#3761AB', '#D0CCB6', '#D95367', '#DAB8A7', '#D3F4E0', '#E38F8B', '#2451A4',
+    '#5681B9', '#A60841', '#BFE1C6', '#C42D52', '#82B3CD', '#F57D58', '#93003A'
+];
+
+const styleCss =
+    `
+    .survey-tbl {
+        width: 100%;
+    }
+    .survey-tbl *{
+        font-family: "GothamBook", "Arial", sans-serif,
+        font-size: 1em
+    }
+    .survey-tbl-r.survey-tbl-head {
+        /* padding-bottom: 0; */
+        background: rgba(218, 230, 237, 0.7);
+        border: #cccccc 1px solid;
+        border-bottom: none;
+    }
+    .survey-sub-tbl .survey-tbl-exp-r:last-of-type {
+        border-bottom: none;
+    }
+    .survey-row-icon {
+        height: 1.5rem;
+        width: 1.5rem;
+        color: #2691D0;
+      }
+    .sub-question-text {
+        text-align: left;
+        font-weight: bold;
+        margin-left: 1em;
+        font-family: GothamBold, Arial, Helvetica, sans-serif;
+    }
+    .survey-row-expanded {
+        margin-top: -1em;
+        border-left: #cccccc 1px solid;
+        border-right: #cccccc 1px solid;
+        background: #f6f6f8;
+    }
+    .active-row {
+        background: #f6f6f8;
+    }
+    .sub-table-1 .sub-question-text, .sub-table-2 .sub-question-text {
+        padding-top: 1rem;
+    }
+
+    .survey-sub-table {
+        padding-left: 2em;
+        padding-bottom: 1em;
+        width: 98%;
+    }
+`;
+
+
+interface SurveyRowProps {
+    surveyConceptId: number;
+    questionConceptId: number;
+    answerConceptId: number;
+    answerValueString: string;
+    hasSubQuestions: string;
+    countValue: number;
+    countPercent: number;
+    isCopeSurvey: boolean;
+    participantCount: number;
+    level: number;
+}
+
+interface SurveyRowState {
+    drawerOpen: boolean;
+    subQuestions: Array<any>;
+    subAnswers: object;
+    subTitle: string;
+}
+
+
+const SurveyAnswerRowComponent = (class extends React.Component<SurveyRowProps, SurveyRowState> {
+    constructor(props: SurveyRowProps) {
+        super(props);
+
+        this.state = {
+            drawerOpen: false,
+            subQuestions: [],
+            subAnswers: {},
+            subTitle: ''
+        };
+    }
+    nextLevel: number;
+
+
+    openDrawer() {
+        this.setState({
+            drawerOpen: !this.state.drawerOpen
+        });
+        if (this.props.hasSubQuestions === '1') {
+            this.nextLevel = this.props.level + 1;
+            this.getSubQuestions();
+        }
```

Comment:
We should add a condition here so we only call the api the first time the drawer opens. Currently, it's calling again every time the drawer opens or closes:
```suggestion
        if (this.props.hasSubQuestions === '1' && !this.state.subQuestions.length) {
            this.getSubQuestions();
        }
```